delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,267 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverRejection._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverState.{CrossProjectCurrent, Current, InProjectCurrent, Initial}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers._
+import fs2.Stream
+import monix.bio.{IO, Task, UIO}
+
+/**
+  * Operations for handling resolvers
+  */
+trait Resolvers {
+
+  /**
+    * Create a new resolver where the id is either present on the payload or self generated
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Create a new resolver with the provided id
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(id: Iri, project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Update an existing resolver
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param rev            the current revision of the resolver
+    * @param resolverFields the payload to update the resolver
+    */
+  def update(id: Iri, project: ProjectRef, rev: Long, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Deprecate an existing resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    * @param rev     the current revision of the resolver
+    */
+  def deprecate(id: Iri, project: ProjectRef, rev: Long)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Fetch the last version of a resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    */
+  def fetch(id: Iri, project: ProjectRef): UIO[Option[ResolverResource]]
+
+  /**
+    * Fetches the resolver at a given revision
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    * @param rev     the current revision of the resolver
+    */
+  def fetchAt(id: Iri, project: ProjectRef, rev: Long): IO[RevisionNotFound, Option[ResolverResource]]
+
+  /**
+    * A non terminating stream of events for resolvers. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def events(offset: Offset = NoOffset): Stream[Task, Envelope[ResolverEvent]]
+
```

Comment:
add another for events for the current project:
```scala
def events(
      projectRef: ProjectRef,
      offset: Offset
  ): IO[WrappedProjectRejection, Stream[Task, Envelope[ResolverEvent]]]
```

Reply:
I don't think we're using that in 1.4.x series, do you think it's useful?

Reply:
I think it is: v1/resolvers/{org}/{proj}/events

But yeah, if we don't need it we can skip it

Reply:
we don't have that according to the docs